packages/nuxt/src/core/nitro.ts

```diff
@@ -524,16 +525,20 @@ function relativeWithDot (from: string, to: string) {
   return relative(from, to).replace(/^([^.])/, './$1') || '.'
 }
 
-function spaLoadingTemplatePath (nuxt: Nuxt) {
-  return typeof nuxt.options.spaLoadingTemplate === 'string'
-    ? resolve(nuxt.options.srcDir, nuxt.options.spaLoadingTemplate)
-    : resolve(nuxt.options.srcDir, 'app/spa-loading-template.html')
+async function spaLoadingTemplatePath (nuxt: Nuxt) {
+  if (typeof nuxt.options.spaLoadingTemplate === 'string') {
```

Comment:
@danielroe Nice changes. I removed this part on my last comment because `nuxt.options._layers` has project's initial config (same as `nuxt.options`) as first item and thought looping through `_layers` would bring same output.

Reply:
You're quite right, it would have a similar effect.

But in this case, for consistency, I think we want the behaviour to be the same as other layer options (like error.vue). We will resolve the path automatically if it's _not_ specified in config, and otherwise let `c12` resolve it for us. (The edge case we avoid here would be something like a more prioritised layer specifying an _invalid_ path and it falling back without an error to the lower-level layer.)